data/json/materials.json

```diff
@@ -2584,14 +2584,21 @@
     "id": "charcoal",
     "name": "Charcoal",
     "copy-from": "wood",
-    "fuel_data": { "energy": "1600 kJ" }
+    "fuel_data": { "energy": "6240 kJ" },
+    "//": "physical density for charcoal estimated to sit at about 0.208 kg/liter, while energy density per kg is slightly lower than pure anthracite coal at 30 MJ/kg.",
+    "//1": "208g of charcoal (1 liter) yields 6240 kJ (30MJ x .208), and a single chunk (8mL/1.664g) is worth 50kJ",
+    "//2": "density sourced from https://www.aqua-calc.com/page/density-table/substance/charcoal and http://ukrfuel.com/news-physical-properties-of-charcoal-22.html",
+    "density": 0.208
   },
   {
     "type": "material",
     "id": "coal",
     "name": "Coal",
     "copy-from": "wood",
-    "fuel_data": { "energy": "30000 kJ" }
+    "//": "mostly pure anthracite coal, 100% purity anthracite has 33.5MJ/liter.",
+    "//1": "33.5MJx1.5 gives energy per liter, or 50kJ in a single 1ml/1.5g unit of coal.",
+    "fuel_data": { "energy": "50000 kJ" },
```

Comment:
These three lines still do not make sense.

Either L2598 is correct, in which case it should be
```
    "//": "mostly pure anthracite coal, 100% purity anthracite has 33.5MJ/liter.",
    "fuel_data": { "energy": "33500 kJ" },
```

Or [your comment here](https://github.com/CleverRaven/Cataclysm-DDA/pull/62180/files/9de4e565f39bc8512d57277fc14b6b78e82db5f0#r1096588040) is correct, i.e.
> energy density of 33.5 kj per gram

in which case it should be
```
    "//": "mostly pure anthracite coal, 100% purity anthracite has 33.5kJ/g.",
    "//1": "33.5kJ/g x 1.5g = 50kJ in a single 1ml/1.5g unit of coal.",
    "fuel_data": { "energy": "50000 kJ" },
```

Reply:
https://github.com/CleverRaven/Cataclysm-DDA/blob/03f25dcc19e8e27f765f768ef56cbd55dc6bbd29/doc/JSON_INFO.md
 "fuel_data" is the energy per liter of fuel. 1ml of coal weighs 1.5 grams, so it the listed fuel per kilogram is multiplied to result in the given fuel per liter of coal because there are 1500 grams of coal per liter of coal.
 I don't think there's anything that needs to be rectified.

Reply:
FFS, read what you wrote at the very end of L2598.

Reply:
I'm a dumbass. I can't believe I didn't notice that. OK, I see the error now. Thanks for persisting.